ietf/draft-ietf-jmap-calendars-18.txt

```diff
@@ -0,0 +1,3360 @@
+
+   The server should create a CalendarEventNotification whenever an
+   event is added, updated or destroyed by another user or due to
+   receiving an iTIP [RFC5546] or other scheduling message in a calendar
+   this user is subscribed to.  The server SHOULD NOT create
+   notifications for events implicitly deleted due to the containing
+   calendar being deleted.
+
+   The CalendarEventNotification does not have any per-user data.  A
+   single instance may therefore be maintained on the server for all
+   sharees of the notification.  The server need only keep track of
+   which users have yet to destroy the notification.
+
+7.1.  Auto-deletion of Notifications
+
+   The server MAY limit the maximum number of notifications it will
+   store for a user.  When the limit is reached, any new notification
+   will cause the previously oldest notification to be automatically
+   deleted.
+
+
+
+
+
+
+Jenkins & Douglass       Expires 9 December 2024               [Page 38]
+
+Internet-Draft               JMAP Calendars                    June 2024
+
+
+   The server MAY coalesce events if appropriate, or remove events that
+   it deems are no longer relevant or after a certain period of time.
+   The server SHOULD automatically destroy a notification about an event
+   if the user updates or destroys that event (e.g. if the user sends an
+   RSVP for the event).
+
+7.2.  Object Properties
+
+   The *CalendarEventNotification* object has the following properties:
+
+   *  *id*: Id
+      The id of the CalendarEventNotification.
+   *  *created*: UTCDateTime
+      The time this notification was created.
+   *  *changedBy*: Person
+      Who made the change.  The Person object has the following
+      properties:
+      -  *name*: String
+         The name of the person who made the change.
+      -  *email*: String|null
+         The email of the person who made the change, or null if no
+         email is available.
+      -  *principalId*: Id|null
+         The id of the Principal corresponding to the person who made
+         the change, if any.  This will be null if the change was due to
+         an entity outside of this user's organisation, e.g. an iTIP
+         invitation from an external person.
+      -  *calendarAddress*: String|null
+         The calendarAddress URI of the person who made the change, if
+         any.  This may be set if the change was made due to receving a
+         scheduling message, such as an iTIP message, in addition to
+         changes made by internal users.
+   *  *comment*: String|null
+      Comment sent along with the change by the user that made it. (e.g.
+      COMMENT property in an iTIP message), if any.
+   *  *type*: String
+      This MUST be one of
+      -  created
+      -  updated
+      -  destroyed
+   *  *calendarEventId*: Id
+      The id of the CalendarEvent that this notification is about.
+   *  *isDraft*: Boolean (created/updated only)
+      Is this event a draft?
```

Comment:
How would a draft event get a CalendarEventNotification? I suppose the only case is if the event is on a shared calendar and another user makes a direct change to it?

It shouldn't get any scheduling messages since isDraft means the server does not send out scheduling messages

Reply:
> How would a draft event get a CalendarEventNotification? I suppose the only case is if the event is on a shared calendar and another user makes a direct change to it?

That's right.